### ISSUE TITLE
Explain reference capability matrix in the tutorial

### ISSUE THREAD

**Issue by EpicEric** [ponylang/pony-tutorial#331]
TL;DR: I believe we should add a reference capability matrix section in a friendly way to help people unfamiliar with actor/capability systems get a better intuition on how they work.

----

One resource that I've personally found very useful when learning about Pony capabilities is the reference capability matrix, reproduced below from CLEBSCH et al. 2015:

-- | Deny global read/write aliases | Deny global write aliases | Allow all global aliases
----- | ----- | ----- | -----
**Deny local read/write aliases** | *iso*
**Deny local write aliases** | trn | *val*
**Allow all local aliases** | ref | box | *tag*
-- | (Mutable) | (Immutable) | (Opaque)

This table also appears in other papers, and talks about Pony, and it has some benefits as it helps visualizing some operations and ideas, such as:

- Sendable capabilities (main diagonal)
- Ephemeral subtyping (walking down/to the right)
- Capability recovery (walking up)
- Impossible capabilities (top-right of the matrix is empty)
- etc.

Currently, the tutorial [explicitly enumerates](https://tutorial.ponylang.io/capabilities/reference-capabilities.html) the capabilities based on relevance, and listing individual attributes, without featuring the matrix anywhere. On one hand, you can get a firm grasp of what each capability does on its own, but I find this approach confusing as it's not very clear how each capability relates to one another. In the next sections, the tutorial says how they relate to one another, but a reader -- such as myself -- might still struggle to grasp the idea of [viewpoint adaptation](https://tutorial.ponylang.io/capabilities/combining-capabilities.html) without a solid base.

However, I also believe that the reference capability matrix above can be specially confusing to new users that haven't used actor languages before, as "local" and "global" in terms of references are not clear concepts for people coming from other programming backgrounds.

I've opened this issue to get feedback on how we could add the same visual intuition that the reference capability matrix provides for people that have never worked before with an actor system, or capabilities, or parallel programming, or all three.

**Comment by SeanTAllen** [comment]
@EpicEric I just redid the reference capabilities stuff a little. You up for doing a PR to add this where appropriate?

**Comment by EpicEric** [comment]
I'll work on this.

**Comment by SeanTAllen** [comment]
Awesome.
